pkg/antctl/raw/packetcapture/command_test.go

```diff
@@ -0,0 +1,258 @@
+			Name:      "antrea-agent-1",
+			Namespace: "kube-system",
+		},
+	}
+	pod1 = v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod-1",
+			Namespace: "default",
+		},
+	}
+	pod2 = v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod-2",
+			Namespace: "default",
+		},
+	}
+	k8sClient = k8sfake.NewSimpleClientset(&pod1, &pod2, &antreaAgentPod)
+)
+
+type testPodFile struct{}
+
+func (p *testPodFile) CopyFromPod(ctx context.Context, fs afero.Fs, namespace, name, containerName, srcPath, dstDir string) error {
+	return nil
+}
+
+func TestRun(t *testing.T) {
+	tcs := []struct {
+		name      string
+		option    packetCaptureOptions
+		expectErr string
+	}{
+		{
+			name: "pod-2-pod",
+			option: packetCaptureOptions{
+				source:  srcPod,
+				dest:    dstPod,
+				flow:    "tcp,tcp_src=50060,tcp_dst=80",
+				timeout: testTimeout,
+				number:  testNum,
+			},
+		},
+		{
+			name: "pod-2-ip",
+			option: packetCaptureOptions{
+				source:  srcPod,
+				dest:    ipv4,
+				nowait:  true,
+				timeout: testTimeout,
+				number:  testNum,
+				flow:    "udp,udp_src=1234,udp_dst=1234",
+			},
+		},
+		{
+			name: "invalid-packetcapture",
+			option: packetCaptureOptions{
+				source:  ipv4,
+				dest:    ipv4,
+				flow:    "icmp",
+				timeout: testTimeout,
+				number:  testNum,
+			},
+			expectErr: "error when constructing a PacketCapture CR: one of source and destination must be a Pod",
+		},
+		{
+			name: "invalid timeout settings",
+			option: packetCaptureOptions{
+				source:  srcPod,
+				dest:    dstPod,
+				timeout: 500 * time.Second,
+				number:  testNum,
+			},
+			expectErr: "timeout cannot be longer than 5m0s",
+		},
+		{
+			name: "invalid packet number",
+			option: packetCaptureOptions{
+				source:  srcPod,
+				dest:    dstPod,
+				timeout: testTimeout,
+			},
+			expectErr: "packet number should be larger than 0",
+		},
+	}
+
+	for _, tt := range tcs {
+		t.Run(tt.name, func(t *testing.T) {
+			client := antreafakeclient.NewSimpleClientset()
+			client.PrependReactor("create", "packetcaptures", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				obj := createAction.GetObject().(*v1alpha1.PacketCapture)
+				if tt.expectErr == "" {
+					obj.Status.FilePath = fmt.Sprintf("%s:/tmp/antrea/packages/%s.pcapng", antreaAgentPod.Name, antreaAgentPod.Name)
+					obj.Status.Conditions = []v1alpha1.PacketCaptureCondition{
+						{
+							Type:   v1alpha1.PacketCaptureComplete,
+							Status: metav1.ConditionTrue,
+						},
+					}
+				}
+				return false, obj, nil
+			})
+			getCopier = func(config *rest.Config, client kubernetes.Interface) raw.PodFileCopier {
+				return &testPodFile{}
+			}
+			defer func() {
+				getCopier = getPodFileCopier
+			}()
+			buf := new(bytes.Buffer)
+			err := packetCaptureRun(context.TODO(), buf, nil, k8sClient, client, &tt.option)
+			if tt.expectErr == "" {
+				require.NoError(t, err)
```

Comment:
I think the test would be more comprehensive if you did some partial matching on the contents of `buf` to make sure it includes the expected output (local path to the pcapng file). What do you think?